source/databricks/settlement_report/tests/domain/time_series/test_read_and_filter.py

```diff
@@ -346,3 +351,290 @@ def test_read_and_filter_for_wholesale__when_system_operator__returns_only_time_
+        ],
+    )
+
+    latest_calculations_df = reduce(
+        lambda df1, df2: df1.union(df2),
+        [
+            latest_calculations_factory.create(
+                spark,
+                default_data.create_latest_calculations_data_spec(
+                    calculation_id=calc_id, calculation_type=calc_type
+                ),
+            )
+            for calc_id, calc_type in calculation_id_and_type.items()
+        ],
+    )
+
+    mock_repository = Mock()
+    mock_repository.read_metering_point_time_series.return_value = time_series_df
+    mock_repository.read_latest_calculations.return_value = latest_calculations_df
+
+    # Act
+    actual_df = read_and_filter_for_balance_fixing(
+        period_start=DEFAULT_FROM_DATE,
+        period_end=DEFAULT_TO_DATE,
+        grid_area_codes=[default_data.DEFAULT_GRID_AREA_CODE],
+        energy_supplier_ids=None,
+        resolution=default_data.DEFAULT_RESOLUTION,
+        time_zone=DEFAULT_TIME_ZONE,
+        repository=mock_repository,
+    )
+
+    # Assert
+    actual_calculation_ids = (
+        actual_df.select(DataProductColumnNames.calculation_id).distinct().collect()
+    )
+    assert len(actual_calculation_ids) == 1
+    assert (
+        actual_calculation_ids[0][DataProductColumnNames.calculation_id]
+        == "22222222-9fc8-409a-a169-fbd49479d718"
+    )
+
+
+def test_read_and_filter_for_balance_fixing__when_two_calculations_with_time_overlap__returns_only_latest_calculation_data(
+    spark: SparkSession,
+) -> None:
+    # Arrange
+    day_1 = DEFAULT_FROM_DATE
+    day_2 = day_1 + timedelta(days=1)
+    day_3 = day_1 + timedelta(days=2)
+    day_4 = day_1 + timedelta(days=3)
+    calculation_id_1 = "11111111-9fc8-409a-a169-fbd49479d718"
+    calculation_id_2 = "22222222-9fc8-409a-a169-fbd49479d718"
+    calc_type = CalculationTypeDataProductValue.BALANCE_FIXING
+
+    time_series_df = reduce(
+        lambda df1, df2: df1.union(df2),
+        [
+            time_series_factory.create(
+                spark,
+                default_data.create_time_series_data_spec(
+                    calculation_id=calc_id,
+                    calculation_type=calc_type,
+                    from_date=from_date,
+                    to_date=to_date,
+                ),
+            )
+            for calc_id, from_date, to_date in [
+                (calculation_id_1, day_1, day_3),
+                (calculation_id_2, day_2, day_4),
+            ]
+        ],
+    )
+
+    latest_calculations_df = latest_calculations_factory.create(
+        spark,
+        [
+            default_data.create_latest_calculations_data_spec(
+                calculation_id=calc_id,
+                calculation_type=calc_type,
+                start_of_day=start_of_day,
+            )
+            for calc_id, start_of_day in [
+                (calculation_id_1, day_1),
+                (calculation_id_1, day_2),
+                (calculation_id_2, day_3),
```

Comment:
This would be more intuitive to me:
```suggestion
                (calculation_id_1, day_1),
                (calculation_id_2, day_2),
                (calculation_id_2, day_3),
                (calculation_id_2, day_4),
```
Anticipating that calculation 2 is newer than calculation 1. I also think you are missing day 4.

Reply:
day_4 is exclusive

Reply:
I added a comment stating that it is exclusive